src/web/components/ClientSideTokenGeneration/CstgDomainInputRow.tsx

```diff
@@ -0,0 +1,128 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import clsx from 'clsx';
+import { ChangeEvent, ReactNode, useState } from 'react';
+import { parse } from 'tldts';
+import { useDebouncedCallback } from 'use-debounce';
+
+import { InlineMessage } from '../Core/InlineMessage';
+
+import './CstgDomainInputRow.scss';
+
+type CstgDomainInputRowProps = {
+  onAdd: (newDomain: string) => Promise<void>;
+  onCancel: () => void;
+};
+
+type DomainProps = {
+  isIcann: boolean | null;
```

Comment:
what's Icann?

Reply:
quick google says it stands for Internet Corporation for Assigned Names and Numbers but what's that mean here?

Reply:
So the lib has a public suffix list built-in.  isICANN means the TLD come from ICANN part of the list

The full list is here: https://publicsuffix.org/list/public_suffix_list.dat

Reply:
thanks! makes sense